daemon/network.go

```diff
@@ -45,7 +45,9 @@ func (daemon *Daemon) FindNetwork(idName string) (libnetwork.Network, error) {
 	// 3. match by ID prefix
 	list := daemon.GetNetworksByIDPrefix(idName)
 	if len(list) == 0 {
-		return nil, errors.WithStack(networkNotFound(idName))
+		// The Error type ErrNoSuchNetwork is also used by the controller to try to create a managed network
+		// asking to the manager, for this reason the type cannot be changed
+		return nil, errors.WithStack(libnetwork.ErrNoSuchNetwork(idName))
```

Comment:
This still needs to implement the `NotFound()` interface.
If we can wrap it in an error type that implements both `NotFound` and `Cause()`, then the caller would just need to do `errors.Cause(err).(libnetwork.ErrNoSuchNetwork)`

Reply:
👍 looks like you're describing what I tried to phrase above 😄

Reply:
k will try to look into this, thanks